Lib/test/test_sysconfig.py

```diff
@@ -592,6 +592,73 @@ def test_osx_ext_suffix(self):
         suffix = sysconfig.get_config_var('EXT_SUFFIX')
         self.assertTrue(suffix.endswith('-darwin.so'), suffix)
 
+    def test_always_set_abiflags(self):
+        self.assertIn('ABIFLAGS', sysconfig.get_config_vars())
+        self.assertIsInstance(sysconfig.get_config_var('ABIFLAGS'), str)
+        self.assertIn('abiflags', sysconfig.get_config_vars())
+        self.assertIsInstance(sysconfig.get_config_var('abiflags'), str)
+
+    def test_always_set_py_debug(self):
+        self.assertIn('Py_DEBUG', sysconfig.get_config_vars())
+        Py_DEBUG = sysconfig.get_config_var('Py_DEBUG')
+        self.assertIn(Py_DEBUG, (0, 1))
+        self.assertEqual(Py_DEBUG, support.Py_DEBUG)
+
+    def test_always_set_py_gil_disabled(self):
+        self.assertIn('Py_GIL_DISABLED', sysconfig.get_config_vars())
+        Py_GIL_DISABLED = sysconfig.get_config_var('Py_GIL_DISABLED')
+        self.assertIn(Py_GIL_DISABLED, (0, 1))
+        self.assertEqual(Py_GIL_DISABLED, support.Py_GIL_DISABLED)
+
+    def test_abiflags(self):
+        # If this test fails on some platforms, maintainers should update the
+        # test to make it pass, rather than changing the definition of ABIFLAGS.
+        abiflags = sysconfig.get_config_var('abiflags')
+        ABIFLAGS = sysconfig.get_config_var('ABIFLAGS')
+
+        self.assertIsInstance(abiflags, str)
+        self.assertIsInstance(ABIFLAGS, str)
+        self.assertIn(abiflags, ABIFLAGS)
+
+        valid_abiflags = ('', 't', 'd', 'td')
+        for flags in valid_abiflags:  # senity check
+            self.assertTrue(len(flags) == len(set(flags)), flags)
```

Comment:
I find this sanity check strange. It’s just checking that the 4 values are not miswritten? I think code review is enough here, no need for this code.